fullstop-aws-support/fullstop-aws-cloudtrail-support/src/main/java/org/zalando/stups/fullstop/events/CloudTrailEventSupport.java

```diff
@@ -239,7 +239,17 @@ public static ViolationBuilder violationFor(CloudTrailEvent cloudTrailEvent) {
         return new ViolationBuilder()
                 .withEventId(getEventId(cloudTrailEvent))
                 .withAccountId(getAccountId(cloudTrailEvent))
-                .withRegion(getRegionAsString(cloudTrailEvent));
+                .withRegion(getRegionAsString(cloudTrailEvent))
+                .withUsername(getUsernameAsString(cloudTrailEvent));
+    }
+
+    public static String getUsernameAsString(CloudTrailEvent cloudTrailEvent) {
+
+        if (cloudTrailEvent == null || cloudTrailEvent.getEventData() == null) {
+            return null;
+        }
+
+        return cloudTrailEvent.getEventData().getUserIdentity().getARN();
```

Comment:
is this always present? Otherwise NullPointerExceptions will occur. I would recommend to use java 8 optionals: `Optional.ofNullable(cloudTrailEvent).map(CloudTrailEvent::getEventData).map(EventData::getUserIdentity).map(UserIdentity::getARN).orElseReturn(null)`